architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,296 @@
+    3. If it is a kubernetes resource such as container, the RP interacts with API server to render the resource.
+3. Application RP uses API server to save Radius resources and Async Operations By default, API Server is used as Radius Datastore and Queue.
+
+
+## Threats
+
+#### Impersonating Applications RP could cause information disclosure, DDoS and misuse of cloud resources.
+
+**Description:** 
+If a malicious actor can impersonate Applications RP, requests from UCP will be received by the malicious actor. The malicious actor can also send requests to UCP and access cloud credentials.
+
+**Impact:** 
+All data sent from UCP to Applications RP will be available to the malicious actor, such as payloads of resources. These payloads can disclose information about Application resources. 
+The impersonator would also be able to retreive credentials through UCP. The credentials can be used to create and consume Azure and AWS resources. The impersonator could also update Environment recipe links to use an outdated/ vulnerable version of the resource.
+
+**Mitigations:**
+
+1. Impersonating Applications RP, tampering with the applications rp code and configuration would require access to modify the `radius-system` namespace. Our threat model assumes that the operator has limited access to the `radius-system` namespace using Kubernetes' existing RBAC mechanism.
+
+2. We should add support in Radius to use mTLS as communication protocol between UCP and Applications RP.
+
+**Status:** 
+
+1. Active. All mitigations which make use of kubernetes RBAC are currently active. Operators are expected to secure their cluster and limit access to the `radius-system` namespace. 
+   
+2. Pending. 
+
+#### Malicious user can make arbitrary requests to Applications RP API. 
+
+**Impact**
+
+This can result in unintended application resource changes.
+
+**Mitigation**
+
+Radius should add validation to make sure the incoming request is from UCP. 
+
+**Status**
+
+Pending
+
+#### Sniffing the communication between Applications RP  and Kubernetes API Server / UCP could cause information disclosure 
+
+**Description:** 
+If a malicious actor could sniff communication between the applications RP and the UCP, the actor could modify/replay the packets.
+
+**Impact:** 
+A malicious actor could use the information about the resources and operations in progress. They can also replay the same requests to cause a DDoS or send a modified payload to cause inadvertant changes to application.
+
+**Mitigations**
+
+1. Tampering with Application RP code/ configs would require access to modify the `radius-system` namespace. Our threat model assumes that the operator has limited access to the `radius-system` namespace using Kubernetes' existing RBAC mechanism. It would be impossible for an arbitrary user to gain access to the cluster and capture traffic.
+
+2. We should add support in Radius to use mTLS as communication protocol between UCP and Applications RP.
```

Comment:
what's the plan for this?  all network traffic should be encrypted and all data encrypted in transit and at rest

Reply:
It is listed as pending action item. we plan to use dapr to bring mtls support.